app/kuma-injector/pkg/injector/injector.go

```diff
@@ -237,11 +253,63 @@ func (i *KumaInjector) NewInitContainer(pod *kube_core.Pod) kube_core.Container
 	}
 }
 
-func (i *KumaInjector) NewAnnotations(pod *kube_core.Pod) map[string]string {
-	return map[string]string{
-		metadata.KumaMeshAnnotation:                    metadata.GetMesh(pod), // either user-defined value or default
+func (i *KumaInjector) NewAnnotations(pod *kube_core.Pod) (map[string]string, error) {
```

Comment:
Let's pass `Mesh` resource as a parameter to `NewAnnotations` (to avoid bubbling up errors and simplify testing)